network/records/subnets.go

```diff
@@ -21,41 +22,30 @@ const (
 
 // UpdateSubnets updates subnets entry according to the given changes.
 // count is the amount of subnets, in case that the entry doesn't exist as we want to initialize it
-func UpdateSubnets(node *enode.LocalNode, count int, added []uint64, removed []uint64) ([]byte, error) {
+func UpdateSubnets(node *enode.LocalNode, added []uint64, removed []uint64) (Subnets, bool, error) {
 	subnets, err := GetSubnetsEntry(node.Node().Record())
 	if err != nil && !errors.Is(err, ErrEntryNotFound) {
-		return nil, errors.Wrap(err, "could not read subnets entry")
-	}
-	orig := make([]byte, len(subnets))
-	if len(subnets) == 0 { // not exist, creating slice
-		subnets = make([]byte, count)
-	} else {
-		copy(orig, subnets)
+		return Subnets{}, false, errors.Wrap(err, "could not read subnets entry")
 	}
+
+	orig := subnets
 	for _, i := range added {
 		subnets[i] = 1
 	}
 	for _, i := range removed {
 		subnets[i] = 0
 	}
-	if bytes.Equal(orig, subnets) {
-		return nil, nil
+	if orig == subnets {
+		return Subnets{}, false, nil
 	}
 	if err := SetSubnetsEntry(node, subnets); err != nil {
-		return nil, errors.Wrap(err, "could not update subnets entry")
+		return Subnets{}, false, errors.Wrap(err, "could not update subnets entry")
 	}
-	return subnets, nil
+	return subnets, true, nil
 }
 
 // Subnets holds all the subscribed subnets of a specific node
-type Subnets []byte
-
-// Clone clones the independent byte slice
-func (s Subnets) Clone() Subnets {
-	cp := make([]byte, len(s))
-	copy(cp, s)
-	return cp
-}
+type Subnets [commons.SubnetsCount]byte
```

Comment:
Perhaps clarify what `byte` values will be expected (from what I've gathered we have `commons.SubnetsCount=128` subnets, and we denote with 1 and 0 whether we participate in a certain subnet or not).

Reply:
>  I'm not sure it will improve perf

Well, yeah, I'm pretty sure it's far from being a bottleneck, I just thought about optimizing it more while we are touching it. I'd say it's not just about fitting the cache line; we'd also have fewer loop cycles and faster array copies.  But I'm inclined to think it's not worth spending time at the moment

Reply:
Yup, I wouldn't bother with such a change just now tbh, there seems to be plenty of other improvements to make too :)